Tests/VB/StatementTests.cs

```diff
@@ -755,6 +755,90 @@ void TestMethod()
     End Sub
 End Class");
         }
+        [Fact(Skip = "month As Integer is reduced by Simplifier, and month threated as Date.Month method")]
+        public async Task ForWithDateTimeVariables() {
+            await TestConversionCSharpToVisualBasic(
+@"class TestClass {
+    void TestMethod() {
+        int summary = 0;
+        for (int month = 1; month <= 12; month++) {
+            summary += month;
+        }
+    }
+}",
+@"Friend Class TestClass
+    Private Sub TestMethod()
+        Dim summary = 0
+
+        For month As Integer = 1 To 12
+            summary += month
+        Next
+    End Sub
+End Class");
+        }
+        [Fact(Skip = "Should we rename, or add If True End If block?")]
```

Comment:
My order of preference would be:
1) Rename - most effort to implement, but we do already have a renaming phase that looks at locals
2) If True - Gets it compiling and keeps the loops isolated, but someone would probably come along and manually rename afterwards
3) Remove the Dim keyword in the second case - Gets it compiling but makes the code a bit harder to maintain